eng/common/scripts/Test-Spelling.ps1

```diff
@@ -0,0 +1,83 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $TargetRef = 'master', 
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $CspellConfigPath = "./.vscode/cspell.json"
+)
+
+. $PSScriptRoot/logging.ps1
+
+
+if ((Get-Command npx | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate npx. Install NodeJS (includes npm and npx) https://nodejs.org/en/download/"
+    exit 1
+}
+
+$initialDirectory = Get-Location
+$exitCode = 0
+try { 
+    Set-Location "$PSScriptRoot/../../.."
+
+    # Lists names of files that were in some way changed between the 
+    # current ref and $TargetRef. Excludes files that were deleted to
+    # prevent errors in Resolve-Path
+    Write-Host "git diff --diff-filter=d --name-only $TargetRef"
```

Comment:
It's probably worth adding a `git` command check similar to the `npx` check since this is going to be running from powershell sessions.